### PR TITLE
Closes #516 | Add/Update Dataloader id_newspaper_2018

### LINKED ISSUE #516: Create dataset loader for ID Newspapers 2018
Dataloader name: `id_newspaper_2018/id_newspaper_2018.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?id_newspaper_2018

|     Dataset| id_newspaper_2018 |
|-------------|---|
| Description | This dataset aims to provide open access to the public to thousands of articles in Indonesian from various sources. The articles are provided along with accompanying article metadata, such as source url, date, and title, along with the content of the article itself. Articles were taken over the period of 01 January 2018 to 20 August 2018 from 7 primary sources (Detik, Kompas, Tempo, CNN Indonesia, Sindo, Republika, Poskota). The original dataset also contains data in html format, which includes raw data (images, along with css and ) from the online news website. |
| Subsets | - |
| Languages | ind |
| Tasks | Language Modeling |
| License | Creative Commons Attribution Share Alike 4.0 (cc-by-sa-4.0) |
| Homepage | https://github.com/feryandi/Dataset-Artikel |
| HF URL | https://huggingface.co/datasets/indonesian-nlp/id_newspapers_2018 |
| Paper URL | - |

### PR DESCRIPTION
**Pull request by raileymontalan**
Closes #516

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/id_newspaper_2018/id_newspaper_2018.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+                        "url": data["url"],
+                        "date": data["date"],
+                        "title": data["title"],
+                        "content": data["content"],
+                    }
+                    yield idx, x
+
+                elif self.config.schema == "seacrowd_ssp":
+                    x = {
+                        "id": idx,
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/id_newspaper_2018/id_newspaper_2018.py:144]:
```suggestion
                        "id": str(idx),
```

File: seacrowd/sea_datasets/id_newspaper_2018/id_newspaper_2018.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        Yields examples as (key, example) tuples.
+        """
+        file_paths = []
+        for path, subdirs, files in os.walk(path):
+            for name in files:
+                if name[-5:] == ".json":
+                    file_paths.append(os.path.join(path, name))
+
+        for idx, file_path in enumerate(file_paths):
+            with open(file_path, "r") as file:
```
Comment (elyanah-aco) [inline seacrowd/sea_datasets/id_newspaper_2018/id_newspaper_2018.py:130]: This line didn't work on my machine if the encoding isn't explicitly defined, so let's do that:
```suggestion
            with open(file_path, "r", encoding="utf-8") as file:
```

**Review by elyanah-aco — COMMENTED** [review]
Hi @raileymontalan, thanks for the PR! Works well on my end, just some nits:

**Comment by raileymontalan** [comment]
Hi @elyanah-aco, have resolved the nitpicks in my latest commit. Ready for review, thanks! :)

**Review by MJonibek — APPROVED** [review]
Checked, LGTM. Great work @raileymontalan @elyanah-aco

**Pull request merged**